lib/apiservers/engine/backends/system.go

```diff
@@ -15,31 +15,148 @@
+	if err != nil {
+		log.Errorf("Could not read system memory info: %v", err)
+	}
 
+	// Check if portlayer server is up
+	plClient := PortLayerClient()
+
+	systemStatus := make([][2]string, 1)
+	systemStatus[0][0] = s.ProductName + " Backend Engine"
+	if pingPortlayer(plClient) {
+		systemStatus[0][1] = "RUNNING"
+	} else {
+		systemStatus[0][1] = "STOPPED"
+	}
+
+	// Retrieve info about the VCH from Exec port layer
+	//FIXME: Write the portlayer code to do this
+
+	// Retrieve number of images from storage port layer
+	numImages, err := getImageCount(plClient)
+	if err != nil {
+		log.Info("System.SytemInfo unable to get image count.")
+	}
+
+	// Retieve container status from port layer
+	containerStatus, err := getContainerStatus(plClient)
+	if err != nil {
+		log.Info("System.SytemInfo unable to get global status on containers")
+	}
+
+	// Build up the struct that the Remote API and CLI wants
 	info := &types.Info{
 		Driver:             Driver,
 		IndexServerAddress: IndexServerAddress,
 		ServerVersion:      ServerVersion,
-		Name:               Name,
+		ID:                 s.ProductName,
+		Containers:         containerStatus.count,
+		ContainersRunning:  containerStatus.numRunning,
+		ContainersPaused:   containerStatus.numPaused,
+		ContainersStopped:  containerStatus.numStopped,
+		Images:             numImages,
+		SystemStatus:       systemStatus,
+		Debug:              (os.Getenv("DEBUG") != ""),
+		NGoroutines:        runtime.NumGoroutine(),
+		SystemTime:         time.Now().Format(time.RFC3339Nano),
+		ExecutionDriver:    Driver,
+		LoggingDriver:      "json-file",
+		CgroupDriver:       "N/A",
+		// FIXME: Get this info once we have event listening service
+		//	NEventsListener    int
+		DockerRootDir:    "N/A",
+		ClusterStore:     "",
+		ClusterAdvertise: "",
+		HTTPProxy:        sockets.GetProxyEnv("http_proxy"),
+		HTTPSProxy:       sockets.GetProxyEnv("https_proxy"),
+		NoProxy:          sockets.GetProxyEnv("no_proxy"),
+
+		// These are system related.  Some refer to cgroup info.  Others are
+		// retrieved from the port layer and are information about the resource
+		// pool.
+		// FIXME: update this once we get the resource pool info
+		KernelVersion:   kernelVersion,         //stubbed
+		OperatingSystem: operatingSystem,       //stubbed
+		OSType:          platform.OSType,       //stubbed
+		Architecture:    platform.Architecture, //stubbed
+		NCPU:            runtime.NumCPU(),      //stubbed
+		MemTotal:        meminfo.MemTotal,      //stubbed
+		CPUCfsPeriod:    false,
+		CPUCfsQuota:     false,
+		CPUShares:       false,
+		CPUSet:          false,
+		OomKillDisable:  true,
+		//	MemoryLimit        bool
+		//	SwapLimit          bool
+		KernelMemory: false,
+		//	IPv4Forwarding     bool
+		//	BridgeNfIptables   bool
+		//	BridgeNfIP6tables  bool `json:"BridgeNfIp6tables"`
+	}
+
+	info.Plugins.Network = append(info.Plugins.Network, "N/A")
+	info.Plugins.Volume = append(info.Plugins.Volume, "N/A")
```

Comment:
We should use this to identify the possible `locations` that can be optionally specified for volume create and eventually the free space. These are the map keys in VirtualContainerHostConfigSpec.VolumeLocations.